### PR TITLE
[BROWSEUI] AutoComplete: Close before opening IME candidates

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

The auto-complete window and the IME candidate window were conflicting each other.
JIRA issue: [CORE-19268](https://jira.reactos.org/browse/CORE-19268)

## Proposed changes

- Call `HideDropDown` method when processing `IMN_OPENCANDIDATE` message.

## NOTE

IMM/IME is currently unstable.

## TODO

- [x] Do tests.

## Comparison

WinXP Japanese:
WinXP
Works as expected.

ReactOS (BEFORE):
before
FAILED. Conflicting.

ReactOS (AFTER):
after
Works as expected.

### CODE REVIEW

**Comment by HBelusca** [comment]
How can one then sees the hint-list of files/subdirectories when entering a directory or file name in the address bar?

**Comment by katahiromz** [comment]
@HBelusca Typing the valid absolute path and a backslash will open AutoComplete if the path has any children.

**Comment by HBelusca** [comment]
Ah, so there's still a way to "switch" between the autocomplete list and the IME one.

File: dll/win32/browseui/CAutoComplete.cpp [hunk 1]
```diff
@@ -21,6 +21,7 @@
  */
 
 #include "precomp.h"
+#include <imm.h>
```
Comment (HBelusca) [inline dll/win32/browseui/CAutoComplete.cpp:24]: Is this for `WM_IME_NOTIFY` or `IMN_OPENCANDIDATE` defines? (maybe add a `// For IMN_OPENCANDIDATE` or similar comment, if you feel like it)

**Review by HBelusca — APPROVED** [review]

File: dll/win32/browseui/CAutoComplete.cpp (hunk 1 above)
Comment (katahiromz) [reply dll/win32/browseui/CAutoComplete.cpp:24]:
```suggestion
#include <imm.h> // IMN_OPENCANDIDATE
```

**Pull request merged**